src/include/stir/DiscretisedDensity.inl

```diff
@@ -210,4 +210,100 @@ get_indices_closest_to_physical_coordinates(const CartesianCoordinate3D<float>&
     this->get_indices_closest_to_relative_coordinates(coords - this->get_origin());
 }
 
+template<int num_dimensions, typename elemT>
+void
+DiscretisedDensity<num_dimensions, elemT>::
+swap_axes_based_on_orientation(CartesianCoordinate3D<float>& coordinates,
+                               const PatientPosition patient_position)
+{
+  // STIR coordinates run:
+  // x: left to right as you face the scanner (as seen from the bed side)
+  // y: top to bottom of the scanner
+  // z: running from deep in the scanner out along the bed (as seen from the bed
+  //    side)
+  // std::cerr << patient_position.get_position_as_string() << std::endl;
+  switch (patient_position.get_position()) {
+  case PatientPosition::unknown_position:
+    // If unknown, assume HFS
+  case PatientPosition::HFS:
+    // HFS means currently in patient LPI
+    coordinates.z() *= -1;
+    break;
+
+  case PatientPosition::HFP:
+    // HFP means currently in patient RAI
+    coordinates.x() *= -1;
+    coordinates.y() *= -1;
+    coordinates.z() *= -1;
+    break;
+
+  case PatientPosition::FFS:
+    // FFS means currently in patient RPS
+    coordinates.x() *= -1;
+    break;
+
+  case PatientPosition::FFP:
+    // FFP means currently in patient LAS
+    coordinates.y() *= -1;
+    break;
+
+  default:
+    throw std::runtime_error("Unsupported patient position, can't convert to LPS.");
+  }
+}
+
+template<int num_dimensions, typename elemT>
+CartesianCoordinate3D<float>
+DiscretisedDensity<num_dimensions, elemT>::
+get_LPS_coordinates_for_physical_coordinates(const CartesianCoordinate3D<float>& coords) const
+{
+  CartesianCoordinate3D<float> flip_coords = coords;
```

Comment:
I sort of prefer the following signature but don't mind too much
```C++
CartesianCoordinate3D<float> flip_coords =swap_axes_based_on_orientation(coords, patient_position);
```

Reply:
So do I, shall change